docs/rfcs/internal-telemetry.md

```diff
@@ -0,0 +1,66 @@
+# Defining guidelines for internal telemetry
+
+## Overview
+
+The Collector supports generating internal telemetry
+that assists end users when operating the Collector. So
+far, much of the telemetry has been added to components as
+needed, without guidelines for component authors to provide
+a consistent experience to end users. The goal of this document
+is to:
+
+- describe the naming and attributes currently in use
+- define consistent units + naming for telemetry emitted by the Collector
+- define the process that should be used to configure new metrics
+
+## Out of scope
+
+This document is not intending to dictate when telemetry should be
+emitted by various Collector components. Considering the various types
+of components, this will be better discussed in a future document.
+
+This document is not intending to provide a comprehensive plan for how
+the Collector or the health of telemetry pipelines should be monitored. There
+is an [OpenTelemetry Enhancement Proposal](https://github.com/open-telemetry/oteps/pull/259) that has already started the process
+to provide this information. The information in this document will be updated once
+the OTEP lands to follow its recommendations.
+
+## Internal telemetry properties
```

Comment:
Can we add a requirement that telemetry describing a component is attributed to a specific component instance? If https://github.com/open-telemetry/opentelemetry-collector/issues/10534 lands, we don't need this language, but otherwise I think it's important that attributes are sufficiently detailed.

Reply:
Added the following note:

> telemetry produced by components should include attributes that identify specific instances of the components